spec/classes/redis_spec.rb

```diff
@@ -81,6 +81,23 @@ class { 'redis::globals':
             end
           end
         end
+
+        describe 'with manage_dnf_module true', if: facts[:os]['family'] == 'RedHat' && facts[:os]['release']['major'].to_i >= 8 do
+          let(:pre_condition) do
+            <<-PUPPET
+            class { 'redis::globals':
+              dnf_module_stream => '6',
+            }
+            class { 'redis':
+              manage_package => true,
+            }
+            PUPPET
+          end
+
+          it { is_expected.to compile.with_all_deps }
+          it { is_expected.to contain_package('redis dnf module').with_ensure('6').that_comes_before('Package[redis]') }
```

Comment:
I suspect `Package<|tag == 'puppet-redis'|>` doesn't match the package. A simpler implementation may be to inline `redis::dnfmodule` in `redis::install` and use a regular relation rather than a collector. In that case you can also consider a regular class parameter in `init.pp` rather than one on globals. With SCL globals makes sense (since it affects paths, service names etc) while with DNF modules it's a lot simpler.

Reply:
I took inspiration from the implementation [here[(https://github.com/puppetlabs/puppetlabs-postgresql/commit/8d0efec9fb5a5df42ed64375c71a11d8b6f90b4c) -- do you recall what inspired that or if your thinking around this has simply change to a more standard model?

Reply:
Mostly because the PostgreSQL module is a lot more complex. For example, if you use the https://www.postgresql.org/ RPMs they're packaged very differently from Red Hat's RPMs. Like Debian's default packages, they are co-installable. That means the version comes back in paths like the data directory. As always, there's upsides and downsides to this.

Reply:
I have changed it around to implement this strategy.